### DISCUSSION TITLE
Missing with specific file extentions - .potx and .xltx

### DISCUSSION THREAD

**Discussion by csaAsh** [microsoft-search/pnp-modern-search#2638] [Q&A] [answered]
I've set up various search results to folders and all that is working well but there are various POTX and XLTX files in the folders which are not shown in the search results. 

If I do a standard sharepoint search, they appear, the same is true using classic search. They just don't appear in the web part. 

Is it something I'm doing wrong or not doing OR is it a known issue?

**Comment by wobba** [comment, marked as answer]
If you share your query template it's easier to help out.

**Reply by csaAsh** [reply]
It's a pretty basic query.

Site:"https://<OUR SHAREPOINT SITE>/sites/QualityManagementSystemQMS/Shared%20Documents/Templates/" IsDocument=True

I've also export the full web part properties if that helps.

{
	"documentationLink": "https://microsoft-search.github.io/pnp-modern-search/",
	"selectedLayoutKey": "DetailsList",
	"resultTypes": [],
	"dataSourceProperties": {
		"queryTemplate": "Site:\"<OUR SHAREPOINT SITE>/sites/QualityManagementSystemQMS/Shared%20Documents/Templates/\" IsDocument=True",
		"enableQueryRules": false,
		"enableLocalization": false,
		"refinementFilters": "FileType:equals(\"*\")",
		"selectedProperties": [
			"AuthorOWSUSER",
			"contentclass",
			"ContentTypeId",
			"Created",
			"CreatedBy",
			"DefaultEncodingURL",
			"Filename",
			"FileType",
			"HitHighlightedProperties",
			"HitHighlightedSummary",
			"HtmlFileType",
			"JobTitle",
			"NormListID",
			"NormSiteID",
			"NormUniqueID",
			"NormWebID",
			"owstaxidmetadataalltagsinfo",
			"Path",
			"SiteLogo",
			"SiteTitle",
			"SPSiteUrl",
			"Title",
			"UniqueID",
			"UserName",
			"WorkPhone"
		],
		"resultSourceId": "8413cd39-2156-4e00-b54d-11efd9abdb89",
		"hitHighlightedProperties": "",
		"trimDuplicates": false,
		"sortList": [
			{
				"uniqueId": "0b2f2e61-d0f8-42cf-be96-d23c781d40c2",
				"sortField": "Filename",
				"isDefaultSort": true,
				"sortDirection": null,
				"isUserSort": null,
				"sortFieldDisplayName": null,
				"sortIdx": 1
			}
		]
	},
	"queryTextSource": 1,
	"layoutProperties": {
		"detailsListColumns": [
			{
				"uniqueId": "86a4e687-48ea-4cdf-a6a0-5b5b2d70986c",
				"name": "Filename",
				"value": "<a href=\"{{slot item @root.slots.PreviewUrl}}\" target=\"_blank\" style=\"color: {{@root.theme.semanticColors.link}}\">\n\t{{slot item @root.slots.Filename}}\n</a>",
				"useHandlebarsExpr": true,
				"minWidth": 50,
				"maxWidth": 310,
				"enableSorting": null,
				"valueSorting": null,
				"isResizable": true,
				"isMultiline": null,
				"sortIdx": 1
			},
			{
				"name": "Created",
				"value": "{{getDate (slot item @root.slots.Date) 'LL'}}",
				"useHandlebarsExpr": true,
				"minWidth": "80",
				"maxWidth": "120",
				"enableSorting": false,
				"isMultiline": false,
				"isResizable": true,
				"sortIdx": 2
			},
			{
				"uniqueId": "cf76b1f3-70b6-485f-a27b-0d2ca58807ef",
				"name": "Created By",
				"value": "CreatedBy",
				"useHandlebarsExpr": null,
				"minWidth": 50,
				"maxWidth": 310,
				"enableSorting": null,
				"valueSorting": null,
				"isResizable": true,
				"isMultiline": null,
				"sortIdx": 3
			}
		],
		"isCompact": false,
		"showFileIcon": true,
		"fieldIconExtension": "FileType",
		"enableGrouping": false,
		"groupByField": "",
		"groupsCollapsed": true
	},
	"showSelectedFilters": false,
	"showResultsCount": false,
	"showBlankIfNoResult": false,
	"useMicrosoftGraphToolkit": false,
	"enableTelemetry": true,
	"itemSelectionProps": {
		"allowItemSelection": false,
		"selectionMode": "AsDataFilter",
		"allowMulti": false,
		"valuesOperator": "or"
	},
	"extensibilityLibraryConfiguration": [
		{
			"name": "Default extensibility library",
			"enabled": true,
			"id": "dc4f961b-dbe0-44b4-982d-5776bf99d015"
		}
	],
	"queryModifierConfiguration": [],
	"queryModifierProperties": {
		"endWhenSuccessfull": false
	},
	"selectedVerticalKeys": [
		"2868d513-4332-454e-be41-402b0a58e520"
	],
	"useVerticals": true,
	"paging": {
		"itemsCountPerPage": 20,
		"pagingRange": 5,
		"showPaging": true,
		"hideDisabled": true,
		"hideFirstLastPages": false,
		"hideNavigation": false,
		"useNextLinks": false
	},
	"adaptiveCardsHostConfig": "{\n\t\"fontFamily\": \"Segoe UI, Helvetica Neue, sans-serif\"\n}",
	"layoutRenderType": "Handlebars",
	"dataSourceKey": "SharePointSearch",
	"templateSlots": [
		{
			"slotName": "Title",
			"slotField": "Title",
			"sortIdx": 1
		},
		{
			"slotName": "Path",
			"slotField": "DefaultEncodingURL",
			"sortIdx": 2
		},
		{
			"slotName": "Summary",
			"slotField": "HitHighlightedSummary",
			"sortIdx": 3
		},
		{
			"slotName": "FileType",
			"slotField": "FileType",
			"sortIdx": 4
		},
		{
			"slotName": "PreviewImageUrl",
			"slotField": "AutoPreviewImageUrl",
			"sortIdx": 5
		},
		{
			"slotName": "PreviewUrl",
			"slotField": "AutoPreviewUrl",
			"sortIdx": 6
		},
		{
			"slotName": "Author",
			"slotField": "AuthorOWSUSER",
			"sortIdx": 7
		},
		{
			"slotName": "Tags",
			"slotField": "owstaxidmetadataalltagsinfo",
			"sortIdx": 8
		},
		{
			"slotName": "Date",
			"slotField": "Created",
			"sortIdx": 9
		},
		{
			"slotName": "SiteId",
			"slotField": "NormSiteID",
			"sortIdx": 10
		},
		{
			"slotName": "WebId",
			"slotField": "NormWebID",
			"sortIdx": 11
		},
		{
			"slotName": "ListId",
			"slotField": "NormListID",
			"sortIdx": 12
		},
		{
			"slotName": "ItemId",
			"slotField": "NormUniqueID",
			"sortIdx": 13
		},
		{
			"slotName": "IsFolder",
			"slotField": "ContentTypeId",
			"sortIdx": 14
		},
		{
			"slotName": "contentclass",
			"slotField": "contentclass",
			"sortIdx": 15
		},
		{
			"slotName": "PersonQuery",
			"slotField": "AADObjectID",
			"sortIdx": 16
		},
		{
			"slotName": "UserDisplayName",
			"slotField": "Title",
			"sortIdx": 17
		},
		{
			"slotName": "UserEmail",
			"slotField": "UserName",
			"sortIdx": 18
		},
		{
			"uniqueId": "6f47662c-05c5-409d-a0e1-25494415d0f9",
			"slotName": "Filename",
			"slotField": "Filename",
			"sortIdx": 19
		}
	],
	"inlineTemplateContent": null,
	"externalTemplateUrl": "",
	"verticalsDataSourceReference": "WebPart.544c1372-77ae-40a0-9095-805ae2a0d5e9.776c492e-ed63-494a-ad13-2a756a9b2d01:pnpSearchVerticalsWebPart",
	"title": "Templates",
	"useDynamicFiltering": false,
	"useInputQueryText": false,
	"queryText": {
		"__type": "DynamicProperty",
		"reference": {
			"_reference": "WebPart.544c1372-7e5a-49ec-8db6-812f76c375f2.38ca5581-5d33-4305-be81-a33b1566e002:pnpSearchBoxWebPart",
			"_sourceId": "WebPart.544c1372-7e5a-49ec-8db6-812f76c375f2.38ca5581-5d33-4305-be81-a33b1566e002",
			"_property": "pnpSearchBoxWebPart"
		}
	},
	"selectedItemFieldValue": {
		"__type": "DynamicProperty",
		"value": ""
	},
	"filtersData": {
		"__type": "DynamicProperty",
		"value": {}
	}
}

**Reply by wobba** [reply]
Solution is quite simple :) Not all template files have `IsDocument=True`. You're better off using a `FileType` filter, adding all extensions you want returned. We had a similar issue for Microsoft Search in the Files vertical which is being deployed soon.

`(FileType:doc OR FileType:docx OR FileType:docm OR FileType:dot OR FileType:dotx OR FileType:dotm OR FileType:odt OR FileType:docb OR FileType:wbk OR FileType:rtf OR FileType:ppt OR FileType:pptx OR FileType:pptm OR FileType:pot OR FileType:potx OR FileType:potm OR FileType:ppam OR FileType:ppsx OR FileType:ppsm OR FileType:sldx OR FileType:sldm OR FileType:ppa OR FileType:pps OR FileType:pptm OR FileType:odp OR FileType:powerpoint OR FileType:xls OR FileType:xlsx OR FileType:xlsm OR FileType:xlb OR FileType:xlsb OR FileType:xlc OR FileType:xlm OR FileType:xltm OR FileType:xlam OR FileType:xlt OR FileType:xltx OR FileType:xla OR FileType:xll OR FileType:xlam OR FileType:xlw OR FileType:excel OR FileType:csv OR FileType:odc) OR IsDocument:1`

**Reply by csaAsh** [reply]
Thanks for the quick reply. 

Just gave that a go and it displays the same. Still can't see potx and xlix files. I tried it with just FileType:potx and still nothing. Any other ideas?

**Reply by csaAsh** [reply]
I just noticed the part about the Verticles, I'm using that. I've just tried with a single search box and it's working. So is there a way currently to make this work and use the verticals web part?

**Reply by csaAsh** [reply]
Just did some more testing with your Query Template suggestion it works when using a search box and then populating the result in Search Results. It doesn't work if the search result is just using the query template to display a list of files, regardless of linking it to the verticals widget.

**Reply by csaAsh** [reply]
continued testing and again using your Query Template suggestion it works if I query Site:"https://MY SHAREPOINT SITE/sites/QualityManagementSystemQMS/Shared%20Documents" 

If however, If use the a folder below that SITE:"https://MY SHAREPOINT SITE/sites/QualityManagementSystemQMS/Shared%20Documents/Templates" it doesn't display potx and xltx files. Any ideas?

**Reply by wobba** [reply]
Your Site filter is wrong. Use `path:https://sites/QualityManagementSystemQMS/Shared%20Documents` together with the filetypes for template files. Or use the siteid in a SiteID filter.

See https://learn.microsoft.com/en-us/sharepoint/technical-reference/crawled-and-managed-properties-overview for a listnof known managed properties...or browse the schema yourself via site settings.

**Reply by csaAsh** [reply]
Still couldn't get sub folders to work. Managed a work around as the names in each folder start with the same letters. Thanks for you help.

**Reply by wobba** [reply]
`path:https://sites/QualityManagementSystemQMS/Shared%20Documents` should match all subfolders as well as it's prefix matching. Or you can try adding a wildcard `path:https://sites/QualityManagementSystemQMS/Shared%20Documents/*`
